server/server.py

```diff
@@ -114,231 +99,23 @@ def handle_client_disconnected_message(self, msg):
         # Remove and don't complain if it wasn't there yet
         self.players.pop(msg.clientid, None)
 
+    def send_all_on_server(self, data, game_server):
+        for player in self.find_players_by(game_server=game_server):
+            player.send(data, self)
+
     def handle_client_connected_message(self, msg):
-        self.players[msg.clientid] = PlayerInfo(msg.clientid, msg.clientaddress, msg.clientport)
+        self.players[msg.clientid] = Player(msg.clientid, msg.clientaddress, msg.clientport, UnauthenticatedState)
 
     def handle_client_message(self, msg):
-
-        current_player = self.players[msg.clientid]
-        current_player.lastreceivedseq = msg.clientseq
-
-        def sendmsg(data, clientid=msg.clientid):
-            self.client_queues[clientid].put((data, self.players[clientid].lastreceivedseq))
-
-        def send_all_on_server(data, server):
-            for player in self.find_players_by(server=server):
-                sendmsg(data, player.id)
+        self.player = self.players[msg.clientid]
```

Comment:
I don't want the player for the current message to become part of the server state. If you need it in other methods, add it as a parameter

Reply:
But it looks like you fixed it already in the next commit